### PR TITLE
[2019.2.1] More control on spun test deamons on git pillar tests

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
More control on spun test deamons on git pillar tests

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/54271
https://github.com/saltstack/salt/issues/54187

### Previous Behavior
Daemons were started by salt using states

### New Behavior
Daemons are started by the tests setup functions

### Tests written?
No - Just trying to get more stable behaviour for existing git pillar tests in case of failure

### Commits signed with GPG?

Yes

### CODE REVIEW

File: tests/integration/__init__.py [hunk 1]
```diff
@@ -1112,22 +1133,6 @@ def __exit__(self, type, value, traceback):
             self.smaster_process.terminate()
         except AttributeError:
             pass
-        #salt.utils.process.clean_proc(self.sub_minion_process, wait_for_kill=50)
```
Comment (dwoz) [inline tests/integration/__init__.py]: Do we need this?

**Review by dwoz — CHANGES_REQUESTED** [review]

File: tests/integration/__init__.py (hunk 1 above)
Comment (s0undt3ch) [reply tests/integration/__init__.py]: It has been commented out for quite a while, hence me deleting it.
I'd say, no.

Comment (s0undt3ch) [reply tests/integration/__init__.py]: This code is being submitted separately in https://github.com/saltstack/salt/pull/54263

Comment (dwoz) [reply tests/integration/__init__.py]: OMG, I didn't see that was getting removed. Even though it's clearly red. Guess that was before my coffee. ☕️

**Review by dwoz — APPROVED** [review]

File: tests/integration/__init__.py (hunk 1 above)
Comment (s0undt3ch) [reply tests/integration/__init__.py]: :smile:

**Review by waynew — APPROVED** [review]

**Review by cmcmarrow — APPROVED** [review]

**Pull request merged**